Mage.Sets/src/mage/cards/c/CaseOfTheBurningMasks.java

```diff
@@ -0,0 +1,206 @@
+        Ability initialAbility = new EntersBattlefieldTriggeredAbility(
+                new DamageTargetEffect(3, "it"));
+        initialAbility.addTarget(new TargetOpponentsCreaturePermanent());
+        initialAbility.addTarget(new TargetOpponentsCreaturePermanent());
+        // To solve -- Three or more sources you controlled dealt damage this turn.
+        Condition toSolveCondition = new CaseOfTheBurningMasksCondition();
+        // Solved -- Sacrifice this Case: Exile the top three cards of your library. Choose one of them. You may play that card this turn.
+        Ability solvedAbility = new ConditionalActivatedAbility(new CaseOfTheBurningMasksEffect(),
+                new SacrificeSourceCost(), SolvedSourceCondition.SOLVED).hideCondition();
+
+        this.addAbility(new CaseAbility(initialAbility, toSolveCondition, solvedAbility)
+                .addHint(CaseOfTheBurningMasksHint.instance),
+                new CaseOfTheBurningMasksWatcher());
+    }
+
+    private CaseOfTheBurningMasks(final CaseOfTheBurningMasks card) {
+        super(card);
+    }
+
+    @Override
+    public CaseOfTheBurningMasks copy() {
+        return new CaseOfTheBurningMasks(this);
+    }
+}
+
+class CaseOfTheBurningMasksCondition implements Condition {
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        CaseOfTheBurningMasksWatcher watcher = game.getState().getWatcher(CaseOfTheBurningMasksWatcher.class);
+        return watcher != null && watcher.damagingCountByController(source.getControllerId()) >= 3;
+    }
+
+    @Override
+    public String toString() {
+        return "Three or more sources you controlled dealt damage this turn";
+    }
+}
+
+enum CaseOfTheBurningMasksHint implements Hint {
+    instance;
+
+    @Override
+    public CaseOfTheBurningMasksHint copy() {
+        return this;
+    }
+
+    @Override
+    public String getText(Game game, Ability ability) {
+        Permanent permanent = game.getPermanent(ability.getSourceId());
+        if (permanent == null) {
+            return "";
+        }
+        if (permanent.isSolved()) {
+            return "Case is solved";
+        }
+        int sources = game.getState()
+                .getWatcher(CaseOfTheBurningMasksWatcher.class)
+                .damagingCountByController(ability.getControllerId());
+        StringBuilder sb = new StringBuilder("Case is unsolved. Sources that dealt damage this turn: ");
+        sb.append(sources);
+        sb.append(" (need 3).");
+        if (sources > 2 && game.isActivePlayer(ability.getControllerId())) {
+            sb.append(" Case will be solved at the end step.");
+        }
+        return sb.toString();
+    }
+}
+
+class CaseOfTheBurningMasksWatcher extends Watcher {
+
+    private final Map<UUID, Set<MageObjectReference>> damagingObjects;
+
+    CaseOfTheBurningMasksWatcher() {
+        super(WatcherScope.GAME);
+        this.damagingObjects = new HashMap<>(0);
```

Comment:
this doesn't really matter but I'm curious why you're specifying the initial capacity?